modules/videopress/videopress.php

```diff
@@ -731,6 +732,32 @@ function videopress_shortcode_options( $options ) {
 		return $options;
 	}
 
+	/**
+	 * An override for the attachment url, which returns back the WPCOM videopress original url,
+	 * if it is set to the the objects metadata. this allows us to show the original uploaded
+	 * file on the WPCOM architecture, instead of the locally uplodaded file,
+	 * which doeasn't exist.
+	 *
+	 * @param string $url
+	 * @param int $post_id
+	 *
+	 * @return mixed
+	 */
+	public function update_attachment_url_for_videopress( $url, $post_id ) {
+
+		$post = get_post( $post_id );
```

Comment:
It may be more efficient to call `get_post_mime_type( $post_id )` instead here -- https://codex.wordpress.org/Function_Reference/get_post_mime_type